lib/ssl/test/ssl_api_SUITE.erl

```diff
@@ -3545,6 +3548,33 @@ cipher_listing(Config) when is_list(Config) ->
     Version = ssl_test_lib:protocol_version(Config, tuple),
     length_exclusive(Version) == length_all(Version).
 
+format_error() ->
+    "".
+format_error(Config) when is_list(Config) ->
+    Errors = [{error, enotconn},
+              {error, closed},
+              {options, {keyfile, "TestFileName", {error,enoent}}},
+              {options, {certfile, "TestFileName", {error,enoent}}},
+              {options, {cacertfile, "TestFileName",{error,enoent}}},
+              {options, {option_not_a_key_value_tuple, [foo]}},
+              {options, {insufficient_crypto_support,{'tlsv1.3','tlsv1.1'}}},
+              {options, {protocol, foo}},
+              {option, server_only, alpn_preferred_protocols},
+              {options, {alpn_advertised_protocols, undefined}},
+              {options, missing_version, {'tlsv1.2',{versions, ['tlsv1.1','tlsv1.3']}}},
+              {options, {no_supported_algorithms, {signature_algs,[]}}},
+              {options, {no_supported_signature_schemes, {signature_algs_cert,[]}}}
+             ],
+    Check = fun(Err) ->
+                    Str = ssl:format_error(Err),
+                    io:format("~p => ~s ~n", [Err, Str]),
+                    %% Verify flat string (why?)
```

Comment:
if comment is still valid, elaborate more please.
if not needed, remove.

Reply:
I don't know why we flatten the string in ssl.erl, it just costs performance.